akka-actor/src/main/scala/akka/io/TcpListener.scala

```diff
@@ -60,7 +60,7 @@ private[io] class TcpListener(selectorRouter: ActorRef,
       ret
     } catch {
       case NonFatal(e) ⇒
-        bindCommander ! bind.failureMessage
+        bindCommander ! bind.failureMessage(Some(e))
```

Comment:
The downside of exposing these exceptions is that due to a design deficiency in NIO they are platform dependent (i.e. which precise subtype of IOException is used for which purpose and with which message varies according to operating system and version). This means that we offer an API that is not reliable. We could try to lift it up by knowing all the quirks and interpreting the error messages, turning the real exception into something that makes more sense, but that would be a lot of work (in particular for those systems that we do not have readily available for testing).